### PR TITLE
Basic Radio Functionality

### PR DESCRIPTION
**Pull request by muhajirdev**

### CODE REVIEW

**Comment by mmahalwy** [comment]
#### 📝 Changed routes:


| [**/radio**](https://quran-com-fxdxqdg4b-qurancom.vercel.app/radio) |
|:-:|
|<a href="https://quran-com-fxdxqdg4b-qurancom.vercel.app/radio">Screenshot of /radio</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/1UzmvNq2ucQGCOWx5a_q0TKRNn-oCe3z8lgbxh4Hl_OlM0p83xfKqHoFEJ5ZdTkBFo6dchZ2jnCZ9xg9C5qpmw.png">(view full size)</a> |







Commit 5ae5f08c63299ba85e8b8d125ab1e0c5f9d6d988 (https://quran-com-fxdxqdg4b-qurancom.vercel.app).

File: src/pages/radio.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import ReciterList from 'src/components/Radio/ReciterStationList';
+
+const Radio = () => {
+  const { t } = useTranslation('common');
+  return (
+    <div className={pageStyle.pageContainer}>
+      <div className={pageStyle.flow}>
+        <div className={classNames(pageStyle.flowItem, radioStyle.title)}>{t('listen-now')}</div>
+        <div className={classNames(pageStyle.flowItem, pageStyle.fullWidth)}>
+          <RandomPlaylist />
```
Comment (tifa2UP) [inline src/pages/radio.tsx:18]: CuratedPlayList

File: src/redux/slices/AudioPlayer/state.ts [hunk 2]
```diff
@@ -107,20 +110,41 @@ interface PlayFromInput {
   chapterId: number;
   reciterId: number;
   timestamp?: number;
+  shouldUseRandomTimestamp?: boolean;
```
Comment (tifa2UP) [inline src/redux/slices/AudioPlayer/state.ts:113]: shouldStartFromRandomTimestamp

File: src/redux/slices/radioStation.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { StationState } from 'src/components/Radio/types';
+import { RootState } from 'src/redux/RootState';
+
+const initialState = {} as StationState;
+
+export const radioStation = createSlice({
```
Comment (tifa2UP) [inline src/redux/slices/radioStation.ts:8]: update Redux to persist preferred station

File: src/redux/store.ts [hunk 4]
```diff
@@ -81,6 +82,7 @@ export const rootReducer = combineReducers({
   defaultSettings,
   fontFaces,
   sidebarNavigation,
+  radioStation,
```
Comment (tifa2UP) [inline src/redux/store.ts:85]: radio

File: src/utils/chapter.ts [hunk 5]
```diff
@@ -127,3 +129,8 @@ export const getChapterReadingProgress = (
   currentVerse: number,
   totalNumberOfVerses: number,
 ): number => Math.ceil((currentVerse * 100) / totalNumberOfVerses);
+
+export const TOTAL_QURAN_CHAPTERS = 114;
```
Comment (tifa2UP) [inline src/utils/chapter.ts:133]: QURAN_CHAPTERS_COUNT

File: src/pages/radio.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+import pageStyle from './index.module.scss';
+import radioStyle from './radio.module.scss';
+
+import Footer from 'src/components/dls/Footer/Footer';
+import RandomPlaylist from 'src/components/Radio/CuratedStationList';
+import ReciterList from 'src/components/Radio/ReciterStationList';
+
+const Radio = () => {
+  const { t } = useTranslation('common');
```
Comment (tifa2UP) [inline src/pages/radio.tsx:12]: SEO tags here

**Review by tifa2UP — APPROVED** [review]

**Pull request merged**